### PR TITLE
Add kusto 4

### PR DESCRIPTION
**Pull request by oflipman**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by oflipman** [comment]
@dsgouda @shahabhijeet 
Can some one please take a look at this PR.
I Closed the PR before.

File: src/SDKs/Kusto/Kusto.Tests/Kusto.Tests.csproj [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  <PropertyGroup>
+    <PackageId>Kusto.Tests</PackageId>
+    <Description>Kusto.Tests Class Library</Description>
+    <AssemblyName>Kusto.Tests</AssemblyName>
+    <VersionPrefix>1.0.0-preview</VersionPrefix>
+    <ApplicationIcon />
+    <OutputType>Library</OutputType>
+    <StartupObject />    
+  </PropertyGroup>
+  <!--<PropertyGroup>
```
Comment (dsgouda) [inline src/SDKs/Kusto/Kusto.Tests/Kusto.Tests.csproj]: Please remove commented lines

File: src/SDKs/Kusto/Kusto.Tests/Kusto.Tests.csproj [hunk 2]
```diff
@@ -0,0 +1,35 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.test.reference.props'))" />
+  <PropertyGroup>
+    <PackageId>Kusto.Tests</PackageId>
+    <Description>Kusto.Tests Class Library</Description>
+    <AssemblyName>Kusto.Tests</AssemblyName>
+    <VersionPrefix>1.0.0-preview</VersionPrefix>
+    <ApplicationIcon />
+    <OutputType>Library</OutputType>
```
Comment (dsgouda) [inline src/SDKs/Kusto/Kusto.Tests/Kusto.Tests.csproj]: Remove lines 8-10

File: src/SDKs/Kusto/Kusto.Tests/Kusto.Tests.csproj [hunk 3]
```diff
@@ -0,0 +1,35 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.test.reference.props'))" />
+  <PropertyGroup>
+    <PackageId>Kusto.Tests</PackageId>
+    <Description>Kusto.Tests Class Library</Description>
+    <AssemblyName>Kusto.Tests</AssemblyName>
+    <VersionPrefix>1.0.0-preview</VersionPrefix>
```
Comment (dsgouda) [inline src/SDKs/Kusto/Kusto.Tests/Kusto.Tests.csproj]: Test projects are never published, please replace this line with
```
<Version>1.0.0</Version>
```

File: src/SDKs/Kusto/Management.Kusto/Microsoft.Azure.Management.Kusto.csproj [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    <PackageTags>management;kusto;;;</PackageTags>
+    <PackageReleaseNotes>
+      <![CDATA[
+        This is a public preview release of the Azure Kusto SDK. Included with this release is creating and managing clusters.
+      ]]>
+</PackageReleaseNotes>
+  </PropertyGroup>
+  <PropertyGroup>
+    <TargetFrameworks>net452;netstandard1.4</TargetFrameworks>
+    <AssemblyVersion>1.0.0.0</AssemblyVersion>
```
Comment (dsgouda) [inline src/SDKs/Kusto/Management.Kusto/Microsoft.Azure.Management.Kusto.csproj]: Please remove 20, 21, `AssemblyVersion` and `AssemblyFileVersion` must be specified in the `AssemblyInfo.cs` file

File: src/SDKs/Kusto/Management.Kusto/Generated/SdkInfo_KustoManagementClient.cs [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+  internal static partial class SdkInfo
+  {
+      public static IEnumerable<Tuple<string, string, string>> ApiInfo_KustoManagementClient
+      {
+          get
+          {
+              return new Tuple<string, string, string>[]
+              {
+                new Tuple<string, string, string>("Kusto", "Clusters", "2018-09-07-preview"),
```
Comment (dsgouda) [inline src/SDKs/Kusto/Management.Kusto/Generated/SdkInfo_KustoManagementClient.cs]: If the REST API version is preview, the Nuget packages published must also be preview. Please start the nuget package versioning from `0.9.0-preview`

File: src/SDKs/Kusto/Kusto.Tests/Kusto.Tests.csproj [hunk 6]
```diff
@@ -0,0 +1,35 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.test.reference.props'))" />
+  <PropertyGroup>
+    <PackageId>Kusto.Tests</PackageId>
+    <Description>Kusto.Tests Class Library</Description>
+    <AssemblyName>Kusto.Tests</AssemblyName>
```
Comment (dsgouda) [inline src/SDKs/Kusto/Kusto.Tests/Kusto.Tests.csproj]: Please add the line
```
 <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
``` 
and fix errors if any

**Review by dsgouda — CHANGES_REQUESTED** [review]

File: src/SDKs/Kusto/Kusto.Tests/Kusto.Tests.csproj [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+	<TreatWarningsAsErrors>true</TreatWarningsAsErrors>
+    <VersionPrefix>1.0.0</VersionPrefix>
```
Comment (dsgouda) [inline src/SDKs/Kusto/Kusto.Tests/Kusto.Tests.csproj]: Replace `VersionPrefix` with `Version`

File: src/SDKs/Kusto/Kusto.Tests/Kusto.Tests.csproj [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.Azure.Management.Storage" Version="4.2.0-preview" />    
+    <PackageReference Include="Microsoft.Azure.Management.Network" Version="10.0.0-preview" />
+    <PackageReference Include="Microsoft.Azure.Management.ResourceManager" Version="1.6.0-preview" />
+	<PackageReference Include="Microsoft.Azure.Graph.RBAC" Version="2.2.2-preview" />
```
Comment (dsgouda) [inline src/SDKs/Kusto/Kusto.Tests/Kusto.Tests.csproj]: nit: fix indentation

File: src/SDKs/Kusto/Management.Kusto/Properties/AssemblyInfo.cs [hunk 9]
```diff
@@ -0,0 +1,18 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Reflection;
+using System.Resources;
+
+[assembly: AssemblyTitle("Microsoft Azure Kusto Management Library")]
+[assembly: AssemblyDescription("Provides management functionality for Microsoft Azure Kusto Resources.")]
+
+[assembly: AssemblyVersion("1.0.0.0")]
```
Comment (dsgouda) [inline src/SDKs/Kusto/Management.Kusto/Properties/AssemblyInfo.cs]: `AssemblyVersion` and `AssemblyFileVersion` must be `0.9.0.0`

**Review by dsgouda — CHANGES_REQUESTED** [review]
Left a few comments

**Review by dsgouda — APPROVED** [review]
LGTM

**Pull request merged**